ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -1,71 +1,50 @@
+    '''kundu_chaudhuri_circle_fit with x, y, weights, mask interface'''
+    weights = weights * mask
+    return kundu_chaudhuri_circle_fit(x, y, weights)
+
+def taubin(x, y, weights, mask):
+    '''taubin_circle_fit with x, y, weights, mask interface'''
+    return taubin_circle_fit(x, y, mask)
+
+FIT_METHODS = [kundu_chaudhuri, taubin]
+FIT_METHOD_BY_NAME = {m.__name__: m for m in FIT_METHODS}
+FIT_METHOD_NAMES = list(FIT_METHOD_BY_NAME.keys())
+
+__all__ = ['MuonRingFitter']
+
+class MuonRingFitter(Component):
+    """Different ring fit algorithms for muon rings
+    """
+    fit_method = traits.CaselessStrEnum(
+        FIT_METHOD_NAMES,
+        default_value=FIT_METHOD_NAMES[0]
+    ).tag(config=True)
+
+    def __call__(self, x, y, img, mask):
+        """allows any fit to be called in form of
+            MuonRingFitter(fit_method = "name of the fit")
         """
-        # First calculate the weighted average positions of the pixels
-        sum_weight = np.sum(weight)
-        av_weighted_pos_x = np.sum(x * weight) / sum_weight
-        av_weighted_pos_y = np.sum(y * weight) / sum_weight
-
-        # The following notation is a bit ugly but directly references the paper notation
-        factor = x**2 + y**2
-
-        a = np.sum(weight * (x - av_weighted_pos_x) * x)
-        a_prime = np.sum(weight * (y - av_weighted_pos_y) * x)
-
-        b = np.sum(weight * (x - av_weighted_pos_x) * y)
-        b_prime = np.sum(weight * (y - av_weighted_pos_y) * y)
-
-        c = np.sum(weight * (x - av_weighted_pos_x) * factor) * 0.5
-        c_prime = np.sum(weight * (y - av_weighted_pos_y) * factor) * 0.5
-
-        nom_0 = ((a * b_prime) - (a_prime * b))
-        nom_1 = ((a_prime * b) - (a * b_prime))
-
-        # Calculate circle centre and radius
-        centre_x = ((b_prime * c) - (b * c_prime)) / nom_0
-        centre_y = ((a_prime * c) - (a * c_prime)) / nom_1
-
-        radius = np.sqrt(
-            # np.sum(weight * ((x - centre_x*u.deg)**2 +
-            # (y - centre_y*u.deg)**2)) / # centre * u.deg ???
-            np.sum(weight * ((x - centre_x)**2 + (y - centre_y)**2)) /
-            sum_weight
-        )
+        fit_function = FIT_METHOD_BY_NAME[self.fit_method]
+        radius, center_x, center_y = fit_function(x, y, img, mask)
 
         output = MuonRingParameter()
-        output.ring_center_x = centre_x  # *u.deg
-        output.ring_center_y = centre_y  # *u.deg
-        output.ring_radius = radius  # *u.deg
-        output.ring_phi = np.arctan(centre_y / centre_x)
-        output.ring_inclination = np.sqrt(centre_x ** 2. + centre_y ** 2.)
-        # output.meta.ring_fit_method = "ChaudhuriKundu"
-        output.ring_fit_method = "ChaudhuriKundu"
+        output.ring_center_x = center_x
+        output.ring_center_y = center_y
+        output.ring_radius = radius
+        output.ring_phi = np.arctan2(center_y, center_x)
+        output.ring_inclination = np.sqrt(center_x ** 2. + center_y ** 2.)
```

Comment:
This is not really an inclinitation, right? 

It still is, because how lengths relate to angles but I find still confusing.

Maybe better ring_center_r, ring_center_phi?

Reply:
Should I really modify this container? It was not introduced in this PR .. and honestly ... modifying a container always makes me uneasy .. who knows in how many places it is being used already.

I'd like another opinion on this.

Reply:
But you modified the contents of the container to not have unit degree anymore (lying docs now maybe even?).

Reply:
no you are right, it is not an inclination

Reply:
It is, the center point of the ring is the inclination of the muon to the optical axis of the telescope.

Reply:
ah btw. these two: `ring_phi` and `ring_inclination` only depend on center x and y ... so why do we store them? I though defining these as properties of the muon container would be nice, what do you think?

Reply:
Probably the other way around is more useful. Ring r and phi are more interesting that x and y. Maybe ist just makes sense to keep both for now.

Reply:
I moved these into properties and renamed them in `containers.py` ... which is strictly speaking not part of this PR .. what why not.